contracts/DAITokenWrapper.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity 0.7.1;
+
+
+import "./interfaces/ITokenWrapper.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Pausable.sol";
+
+/**
+ * @title DAITokenWrapper
+ * @notice Contract for wrapping call to DAI token permit function because the DAI token permit function has a different signature from other tokens with which the protocol integrates
+ */
+contract DAITokenWrapper is 
+    ITokenWrapper,
+    Pausable,
+    Ownable
+{
+
+    address private daiTokenAddress;
+    address private bosonRouterAddress;
+  
+
+    constructor(
+        address _daiTokenAddress
+    ) 
+    notZeroAddress(_daiTokenAddress)
+    {
+        daiTokenAddress = _daiTokenAddress;
+        
+    }
+
+    /**
+     * @notice  Checking if a non-zero address is provided, otherwise reverts.
+     */
+    modifier notZeroAddress(address tokenAddress) {
+        require(tokenAddress != address(0), "0A"); //zero address
+        _;
+    }
+
+    /**
+     * @notice Conforms to ERC2612. Calls permit on token, which may or may not have a permit function that conforms to ERC2612
+     * @param owner Address of the token owner who is approving tokens to be transferred by spender
+     * @param spender Address of the party who is transferring tokens on owner's behalf
+     * @param value Number of tokens to be transferred
+     * @param deadline Time after which this permission to transfer is no longer valid
+     * @param v Part of the owner's signatue
+     * @param r Part of the owner's signatue
+     * @param s Part of the owner's signatue
+     */
+    function permit(
+        address owner,
+        address spender,
+        uint256 value,
```

Comment:
This variable is unused. Is the "Number of tokens to be transfered" not needed in the body of this method ?

Reply:
+1 with the reason given by Heather.
However I can't encourage the idea to have to deal with specificity of each token in the BosonRouter contract (or anywhere in the core protocol). Instead we are trying to have the protocol supporting an unbounded list of tokens (all ERC20 compliants), each specificity being dealt in the Wrapper contract of each token